### PR TITLE
Edit Survey - Delete survey button

### LINKED ISSUE #1551: [Edit survey] Ability to delete survey
Requested by @lecrabe 

@vittorino for UX guidance

### PR DESCRIPTION
**Pull request by rfontanarosa**
closes #1551

https://github.com/google/ground-platform/assets/10047945/f70c5bb5-d9e0-426f-8c70-f6d0672fe794

### CODE REVIEW

File: web/src/app/pages/edit-survey/job-dialog/job-dialog.component.html [hunk 1]
```diff
@@ -44,6 +44,16 @@ <h1 mat-dialog-title>{{ title }}</h1>
       All predefined data collection sites and their associated sites will
       be permanently deleted.
     </p>
+
+    <p *ngSwitchCase="DialogType.DeleteOption">
+      Are you sure you wish to delete this option?
+      Any associated data will be lost. This cannot be undone.
```
Comment (gino-m) [inline web/src/app/pages/edit-survey/job-dialog/job-dialog.component.html:50]: Nit: While we're here, I'd replace "any" with "all".

File: web/src/app/pages/edit-survey/job-dialog/job-dialog.component.html [hunk 2]
```diff
@@ -44,6 +44,16 @@ <h1 mat-dialog-title>{{ title }}</h1>
...
+    </p>
+
+    <p *ngSwitchCase="DialogType.DeleteSurvey">
+      Are you sure you wish to delete this survey?
+      Any associated data will be lost. This cannot be undone.
```
Comment (gino-m) [inline web/src/app/pages/edit-survey/job-dialog/job-dialog.component.html:55]: Same here.

File: web/src/app/pages/edit-survey/edit-details/edit-details.component.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      }
+
+      &.delete-survey {
+        display: flex;
+        align-items: center;
+
+        button {
+          border-radius: 100px;
+          border: 1px solid #747775;
+          color: #BA1A1A;
```
Comment (gino-m) [inline web/src/app/pages/edit-survey/edit-details/edit-details.component.scss:49]: Is the there a variable we can refer to here and above for color codes?

File: web/src/app/pages/edit-survey/edit-details/edit-details.component.html [hunk 4]
```diff
@@ -14,9 +14,27 @@
...
+      ></survey-details>
+    </div>
+  </div>
+
+  <div class="survey-detail-card">
+    <div class="card-title">Delete survey</div>
+    <div class="card-row delete-survey">
+      <div style="flex: 1;">Permanently delete survey and associated data</div>
+
+      <button mat-flat-button (click)="openDeleteSurveyDialog()">
```
Comment (gino-m) [inline web/src/app/pages/edit-survey/edit-details/edit-details.component.html:36]: In the video the background looks pink, could just be my eyes are off. Also please check the spacing:

image

**Review by gino-m — CHANGES_REQUESTED** [review]

File: web/src/app/pages/edit-survey/edit-details/edit-details.component.scss (hunk 3 above)
Comment (rfontanarosa) [reply web/src/app/pages/edit-survey/edit-details/edit-details.component.scss:58]: that specific red is used only there

**Review by amysorto — APPROVED** [review]

**Pull request merged**